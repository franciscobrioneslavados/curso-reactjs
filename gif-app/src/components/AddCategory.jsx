import { useState } from 'react'

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('')

  const onInputChange = (target) => {
    setInputValue(event.target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault() // Evita el refresh del navegador
    if (inputValue.trim().length <= 1) return

    //setCategories((categories) => [inputValue, ...categories])
    onNewCategory(inputValue.trim())
    setInputValue('')
  }

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  )
}
