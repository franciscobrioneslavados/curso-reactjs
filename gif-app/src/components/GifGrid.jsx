import { GiftItem } from './GiftItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category)

  return (
    <>
      <h4>{category}</h4>
      <div className="card-grid">
        {images.map((image) => (
          <GiftItem key={image.id} {...image} />
        ))}
      </div>
    </>
  )
}
