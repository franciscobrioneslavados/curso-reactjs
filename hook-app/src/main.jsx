import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
import { CounterApp } from './useState/CounterApp';
import { CounterWithCustomHookApp } from './useState/CounterWithCustomHook';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterWithCustomHookApp />
  </React.StrictMode>,
)
