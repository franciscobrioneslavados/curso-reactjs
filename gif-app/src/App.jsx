import { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

export const App = () => {
  const [categories, setCategories] = useState(['One Punch'])

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return

    // setCategories([...categories, 'Valorant']); // 1era forma de manejar estados
    setCategories((cat) => [newCategory, ...cat]) // 2da forma de manejar estados
  }

  return (
    <>
      <h1>Gif App</h1>

      <AddCategory
        // setCategories={setCategories}
        onNewCategory={(event) => onAddCategory(event)}
      />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  )
}
