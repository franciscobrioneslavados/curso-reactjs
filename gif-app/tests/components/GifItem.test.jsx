import { render, screen } from '@testing-library/react'
import { GiftItem } from '../../src/components'

describe('Pruebas en <GifItem />', () => {
  const title = 'Saitama'
  const url = 'https://saitama.com'

  test('debe de hacer match con el snapshot', () => {
    const { container } = render(<GiftItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })

  test('debe de mostrar la imagen con el url y el alt indicado', () => {
    render(<GiftItem title={title} url={url} />)
    // screen.debug();
    // expect(screen.getByRole('img').src).toBe(url + "/")
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url + '/')
    expect(alt).toBe(title)
  })

  test('debe de mostrar el titulo en el componente', () => {
    render(<GiftItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy();
  })
})
