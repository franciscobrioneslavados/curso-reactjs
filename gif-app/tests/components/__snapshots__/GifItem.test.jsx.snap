// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de jacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://saitama.com"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;

exports[`Pruebas en <GifItem /> debe de jacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://saitama.com"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
